---
title: "Soamtic_mutation"
author: "Subinoy Biswas"
date: "9/23/2019"
output:
  html_document:
    df_print: paged
---

```{r, message=FALSE,echo=FALSE}
library(maftools)
#Requires BSgenome object
library(BSgenome.Hsapiens.UCSC.hg19, quietly = TRUE)
library('NMF')
```

### Taking publicly available dataset from Esophageal carcinoma, acute myeloid leukemia and breast invasive carcinoma the following analysis has been done. I try to include as much results possible as I can.


```{r, message=FALSE,echo=FALSE}



#path to TCGA LAML MAF file
laml.maf = system.file('extdata', 'tcga_laml.maf.gz', package = 'maftools') 

laml.clin = system.file('extdata', 'tcga_laml_annot.tsv', package = 'maftools') 

laml = read.maf(maf = laml.maf, clinicalData = laml.clin)
```

```{r, message=FALSE, echo=FALSE}



laml

#Showing sample summry.
getSampleSummary(laml)

#Showing gene summary.
getGeneSummary(laml)

#showing clinical data associated with samples
getClinicalData(laml)


```

#### summary of the maf file, which displays number of variants in each sample as a stacked barplot and variant types as a boxplot summarized by Variant_Classification

```{r, message=FALSE, echo=FALSE}

plotmafSummary(maf = laml, rmOutlier = TRUE, addStat = 'median',
               dashboard = TRUE, titvRaw = FALSE)

```
#### This is oncoplots, also known as waterfall plots.

```{r}



#oncoplot for top ten mutated genes.
oncoplot(maf = laml, top = 10)



```


#### Mutations in each sample 

```{r, message=FALSE,echo=FALSE}
oncostrip(maf = laml, genes = c('DNMT3A','NPM1', 'RUNX1'))
```
#### Classifies SNPs into Transitions and Transversions and returns a list of summarized tables in various ways

##### boxplot showing overall distribution of six different conversions and as a stacked barplot showing fraction of conversions in each sample.
```{r, message=FALSE,echo=FALSE}
laml.titv = titv(maf = laml, plot = FALSE, useSyn = TRUE)
#plot titv summary
plotTiTv(res = laml.titv)
```


#### Lollipop plots are simple and most effective way showing mutation spots on protein structure. Many oncogenes have a preferential sites which are mutated more often than any other locus. These spots are considered to be mutational hot-spots and lollipop plots can be used to display them along with rest of the mutations.
```{r, message=FALSE, echo=FALSE}
#lollipop plot for DNMT3A, which is one of the most frequent mutated gene in Leukemia.
lollipopPlot(maf = laml, gene = 'DNMT3A', AACol = 'Protein_Change', 
             showMutationRate = TRUE)

lollipopPlot(maf = laml, gene = 'KIT', AACol = 'Protein_Change',
             labelPos = 816, refSeqID = 'NM_000222')

```

#### Cancer genomes, especially solid tumors are characterized by genomic loci with localized hyper-mutations 5. Such hyper mutated genomic regions can be visualized by plotting inter variant distance on a linear genomic scale. These plots generally called rainfall plots 

```{r, message=FALSE,echo=FALSE}
brca <- system.file("extdata", "brca.maf.gz", package = "maftools")
brca = read.maf(maf = brca, verbose = FALSE)

rainfallPlot(maf = brca, detectChangePoints = TRUE, pointSize = 0.6)


```


```{r, message=FALSE,echo=FALSE, include=FALSE}



all.lesions <- system.file("extdata", "all_lesions.conf_99.txt", package = "maftools")
amp.genes <- system.file("extdata", "amp_genes.conf_99.txt", package = "maftools")
del.genes <- system.file("extdata", "del_genes.conf_99.txt", package = "maftools")
scores.gis <- system.file("extdata", "scores.gistic", package = "maftools")

laml.gistic = readGistic(gisticAllLesionsFile = all.lesions, 
                         gisticAmpGenesFile = amp.genes, 
                         gisticDelGenesFile = del.genes, 
                         gisticScoresFile = scores.gis,
                         isTCGA = TRUE)



```

#### Genome plots

```{r, message=FALSE,echo=FALSE}
#GISTIC object
laml.gistic

gisticChromPlot(gistic = laml.gistic, markBands = "all")
```


#### Bubble plot
```{r, message=FALSE,echo=FALSE}
gisticBubblePlot(gistic = laml.gistic)
```


#### Oncoplot sorted according to FAB classification

```{r, message=FALSE,echo=FALSE}
gisticOncoPlot(gistic = laml.gistic, clinicalData = getClinicalData(x = laml),
               clinicalFeatures = 'FAB_classification', 
               sortByAnnotation = TRUE, top = 10)


tcga.ab.009.seg <- system.file("extdata", "TCGA.AB.3009.hg19.seg.txt", package = "maftools")
plotCBSsegments(cbsFile = tcga.ab.009.seg)

```

#### Many disease causing genes in cancer are co-occurring or show strong exclusiveness in their mutation pattern.  Such mutually exclusive or co-occurring set of genes were detected by pair-wise Fisherâ€™s Exact test
 
```{r, message=FALSE,echo=FALSE}
#exclusive/co-occurance event analysis on top 10 mutated genes. 
somaticInteractions(maf = laml, top = 25, pvalue = c(0.05, 0.1))
```


```{r, message=FALSE,echo=FALSE}
oncostrip(maf = laml, genes = c('TP53', 'FLT3', 'RUNX1'))
```

#### Detecting cancer driver genes based on positional clustering
```{r, message=FALSE,echo=FALSE}
laml.sig = oncodrive(maf = laml, AACol = 'Protein_Change',
                     minMut = 5, pvalMethod = 'zscore')

head(laml.sig)

plotOncodrive(res = laml.sig, fdrCutOff = 0.1, useFraction = TRUE)
```

#### Adding and summarizing pfam domains

```{r, message=FALSE,echo=FALSE}
laml.pfam = pfamDomains(maf = laml, AACol = 'Protein_Change', top = 10)

#Protein summary (Printing first 7 columns for display convenience)
laml.pfam$proteinSummary[,1:7, with = FALSE]

#Domain summary (Printing first 3 columns for display convenience)
laml.pfam$domainSummary[,1:3, with = FALSE]
```

#### Pan-Cancer comparison
```{r, message=FALSE,echo=FALSE}
#MutsigCV results for TCGA-AML
laml.mutsig <- system.file("extdata", "LAML_sig_genes.txt.gz", package = "maftools")
pancanComparison(mutsigResults = laml.mutsig, qval = 0.1,
                 cohortName = 'LAML', 
                 inputSampleSize = 200, 
                 label = 1)

```

#### Survival analysis
```{r, message=FALSE,echo=FALSE}
#Survival analysis based on grouping of DNMT3A mutation status
mafSurvival(maf = laml, genes = 'DNMT3A', 
            time = 'days_to_last_followup',
            Status = 'Overall_Survival_Status', isTCGA = TRUE)

#Primary APL MAF
primary.apl = system.file("extdata", "APL_primary.maf.gz", package = "maftools")
primary.apl = read.maf(maf = primary.apl)
#Relapse APL MAF
relapse.apl = system.file("extdata", "APL_relapse.maf.gz", package = "maftools")
relapse.apl = read.maf(maf = relapse.apl)



```



```{r, fig.height=10, fig.width=12, message=FALSE,echo=FALSE}
#Considering only genes which are mutated in at-least in 5 samples in one of the cohort to avoid bias due to genes mutated in single sample.
pt.vs.rt <- mafCompare(m1 = primary.apl, m2 = relapse.apl, m1Name = 'Primary', m2Name = 'Relapse', minMut = 5)
print(pt.vs.rt)

forestPlot(mafCompareRes = pt.vs.rt, pVal = 0.1, 
           color = c('royalblue', 'maroon'), geneFontSize = 0.8)

genes = c("PML", "RARA", "RUNX1", "ARID1B", "FLT3")
coOncoplot(m1 = primary.apl, m2 = relapse.apl, m1Name = 'PrimaryAPL',
           m2Name = 'RelapseAPL', genes = genes, removeNonMutated = TRUE)

lollipopPlot2(m1 = primary.apl, m2 = relapse.apl, gene = "PML", 
              AACol1 = "amino_acid_change", 
              AACol2 = "amino_acid_change", 
              m1_name = "Primary", m2_name = "Relapse")



```
```{r, message=FALSE,echo=FALSE}
fab.ce = clinicalEnrichment(maf = laml, clinicalFeature = 'FAB_classification')

#Results are returned as a list. Significant associations p-value < 0.05
fab.ce$groupwise_comparision[p_value < 0.05]

plotEnrichmentResults(enrich_res = fab.ce, pVal = 0.05)

dgi = drugInteractions(maf = laml, fontSize = 0.75)


dnmt3a.dgi = drugInteractions(genes = "DNMT3A", drugs = TRUE)



```


```{r, message=FALSE,echo=FALSE}

#Printing selected columns.
dnmt3a.dgi[,.(Gene, interaction_types, drug_name, drug_claim_name)]

OncogenicPathways(maf = laml)


PlotOncogenicPathways(maf = laml, pathways = "RTK-RAS")

```



```{r eval=FALSE, include=FALSE}

#Heterogeneity in sample TCGA.AB.2972
tcga.ab.2972.het = inferHeterogeneity(maf = laml, tsb = 'TCGA-AB-2972',
                                      vafCol = 'i_TumorVAF_WU')

print(tcga.ab.2972.het$clusterMeans)

#Visualizing results
plotClusters(clusters = tcga.ab.2972.het)

```


```{r eval=FALSE, include=FALSE}

seg = system.file('extdata', 'TCGA.AB.3009.hg19.seg.txt', package = 'maftools')
tcga.ab.3009.het = inferHeterogeneity(maf = laml, 
                                      tsb = 'TCGA-AB-3009', 
                                      segFile = seg, 
                                      vafCol = 'i_TumorVAF_WU')

#Visualizing results. Highlighting those variants on copynumber altered variants.
plotClusters(clusters = tcga.ab.3009.het, genes = 'CN_altered', showCNvars = TRUE)


```


```{r eval=FALSE, include=FALSE}

laml.tnm = trinucleotideMatrix(maf = laml, prefix = 'chr', 
                               add = TRUE, ref_genome = "BSgenome.Hsapiens.UCSC.hg19")


plotApobecDiff(tnm = laml.tnm, maf = laml, pVal = 0.2)

```

```{r eval=FALSE, include=FALSE}
laml.sign = extractSignatures(mat = laml.tnm, nTry = 3, plotBestFitRes = FALSE)


plotSignatures(laml.sign, title_size = 0.8, )

plotSignatures(laml.sign, title_size = 0.8, )
```

